src/backend/distributed/utils/shardinterval_utils.c

```diff
@@ -460,12 +459,12 @@ SingleReplicatedTable(Oid relationId)
 		return false;
 	}
 
-	/* checking only for the first shard id should suffice */
-	Oid shardId = (*(uint64 *) linitial(shardList));
```

Comment:
`Oid` is 32 bits. We shouldn't be truncating shard ids. This should be the only observable change in this commit